src/lib/y2network/interface_config_builders/br.rb

```diff
@@ -0,0 +1,71 @@
+require "yast"
+require "y2network/config"
+require "y2network/interface_config_builder"
+
+module Y2Network
+  module InterfaceConfigBuilders
+    class Br < InterfaceConfigBuilder
+
+      include Yast::Logger
+
+      def initialize
+        super(type: "br")
+      end
+
+      # @return [Array<Interface>] list of interfaces usable in bridge
+      def select_bridgeable
+        interfaces.all.select { |i| bridgeable?(i) }
+      end
+
+    private
+
+      def interfaces
+        Config.find(:yast).interfaces
+      end
+
+      # Checks whether an interface can be bridged in particular bridge
+      #
+      # @param iface [Interface] an interface to be validated as the bridge slave
+      def bridgeable?(iface)
+        # cannot enslave itself
+        # FIXME: this can happen only bcs we silently use LanItems::Items which
+        # already contains partially configured bridge when adding
+        return false if iface.name == @name
+        return true if !iface.configured
+
+        if interfaces.bond_index[iface.name]
+          log.debug("Excluding (#{iface.name}) - is bonded")
+          return false
+        end
+
+        # the iface is already in another bridge
+        if interfaces.bridge_index[iface.name] && interfaces.bridge_index[iface.name] != @name
```

Comment:
btw those methods just show how strange is to know in collection attribute of interface. Ideally api should look like
```
if iface.bridge && iface.bridge.name != @name
```

Reply:
after all it still is only conversion of old code into new environment. I expect a lot of space for improvements.